pkg/apiserver/rest/dualwriter_mode2.go

```diff
@@ -316,32 +313,32 @@ func (d *DualWriterMode2) Compare(storageObj, legacyObj runtime.Object) bool {
 
 func parseList(legacyList []runtime.Object) (metainternalversion.ListOptions, map[string]int, error) {
 	options := metainternalversion.ListOptions{}
-	originKeys := []string{}
 	indexMap := map[string]int{}
-
-	for i, obj := range legacyList {
-		metaAccessor, err := utils.MetaAccessor(obj)
-		if err != nil {
-			return options, nil, err
-		}
-		originKeys = append(originKeys, metaAccessor.GetOriginKey())
-
-		accessor, err := meta.Accessor(obj)
-		if err != nil {
-			return options, nil, err
-		}
-		indexMap[accessor.GetName()] = i
-	}
-
-	if len(originKeys) == 0 {
-		return options, nil, nil
-	}
-
-	r, err := labels.NewRequirement(utils.AnnoKeyOriginKey, selection.In, originKeys)
-	if err != nil {
-		return options, nil, err
-	}
-	options.LabelSelector = labels.NewSelector().Add(*r)
+	//originKeys := []string{}
```

Comment:
@leonorfmartins -- is this used?  I don't quite understand the intended behavior

Using origin would be great for provisioning, but I do not think it has any role in the dual write strategies.  

We are currently shoving playlist on origin fields just for legacy conversion needs -- but nothing that can help synchronization.

Reply:
The dual writer is/was using it to store the legacy id of the objects it creates, treating the legacy db as the "origin"

Reply:
I get that it is saving it -- that makes total sense.  

The part I am not sure is why it is querying for them.  Both systems should have the same name (uid) -- so if we are missing something, query for name?